src/sentry/sentry_metrics/consumers/indexer/batch.py

```diff
@@ -71,16 +75,27 @@ def invalid_metric_tags(tags: Mapping[str, str]) -> Sequence[str]:
     return invalid_strs
 
 
+def extract_use_case_id(mri: str) -> Optional[UseCaseId]:
+    """
+    Returns the use case ID given the MRI, returns None if MRI is invalid.
+    """
+    use_case_mapping = {"transactions": "performance", "sessions": "release-health"}
```

Comment:
I feel we should have a new abstraction which does this mapping. The current method leads the reader to believe that release health data, performance data, <other registered data> can all flow through both release health and metrics pipeline which is not true. We could have different abstractions for generic metrics and release health and apply the correct abstraction based on the pipeline.

Some of the indexer differences are arising out of need for supporting different use cases in generic metrics platform which is not needed in release health pipeline. If we can isolate these differences in another abstraction it would be easier to make sense of the code.